spec/jobs/collection_upload_event_jobs_spec.rb

```diff
@@ -83,6 +84,35 @@
       expect(follower3.events.length).to eq(0)
     end
   end
+  
+  context 'with follower of child and parent collection' do
+    let(:child) { make_collection(
+      user,
+      title: "Lil' Hamlet",
+      id: 'test-321',
+      visibility: 'open'
+    ) }
+      
+    before do
+      collection.visibility = 'open'
+      child.set_follower(follower4)
+      collection.set_follower(follower4)
+      child.save!
+      collection.save!
+    end
+    
+    it 'does not process duplicate events for child and parent followers' do
+      expect(Time).to receive(:now).at_least(:once).and_return(1)
+
+      expect(job).to receive(:log_for_collection_follower).with(
+        job.collection
+      ).and_call_original
+      expect(job).to receive(:log_to_followers).and_call_original
+      job.run
+      
+      expect(follower4.events.length).to eq(1)
+    end
+  end
```

Comment:
Needs multi-parent tests with scenarios where child's parents are not being followed by current user.